### PR TITLE
use test preparer

### PR DESCRIPTION
**Pull request by xiangyan99**

### CODE REVIEW

**Comment by xiangyan99** [comment]
#9062

File: sdk/appconfiguration/azure-appconfiguration/tests/app_configuration_client_async_preparer.py [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+        )
+        self.resource_group_parameter_name = resource_group_parameter_name
+        self.aad_mode = aad_mode
+
+    def _get_resource_group(self, **kwargs):
+        try:
+            return kwargs[self.resource_group_parameter_name]
+        except KeyError:
+            template = (
+                "To create a key vault a resource group is required. Please add "
```
Comment (lmazuel) [inline sdk/appconfiguration/azure-appconfiguration/tests/app_configuration_client_async_preparer.py]: copy/paste from KeyVault :)

File: sdk/appconfiguration/azure-appconfiguration/tests/app_configuration_preparer.py [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+        )
+        self.resource_group_parameter_name = resource_group_parameter_name
+        self.aad_mode = aad_mode
+
+    def _get_resource_group(self, **kwargs):
+        try:
+            return kwargs[self.resource_group_parameter_name]
+        except KeyError:
+            template = (
+                "To create a key vault a resource group is required. Please add "
```
Comment (lmazuel) [inline sdk/appconfiguration/azure-appconfiguration/tests/app_configuration_preparer.py]: Same

File: sdk/appconfiguration/azure-appconfiguration/tests/test_azure_configuration_client_async.py [hunk 3]
```diff
@@ -31,28 +31,29 @@
 import os
 import logging
 import re
+import hashlib
+from app_configuration_preparer import AppConfigurationPreparer
+from app_configuration_client_async_preparer import AppConfigurationClientPreparer
 
 class AppConfigurationClientTest(AzureMgmtTestCase):
+    try:
+        name_prefix = "appconfig-" + hashlib.md5(os.environ['RUN_IDENTIFIER'].encode()).hexdigest()[-3:]
```
Comment (lmazuel) [inline sdk/appconfiguration/azure-appconfiguration/tests/test_azure_configuration_client_async.py]: Don't do that, use the random_name_enable instead in ALL your RG preparer:
https://github.com/Azure/azure-sdk-for-python/blob/2c9b3309648c2ba669ac04ac318eef2ea15632e2/sdk/keyvault/azure-keyvault-secrets/tests/test_secrets_async.py#L65

If your AppConfiguration mgmt has the same kind of randmness needed (like a storage account must be unique worldwide), please force it in your __init__:
https://github.com/Azure/azure-sdk-for-python/blob/2c9b3309648c2ba669ac04ac318eef2ea15632e2/sdk/keyvault/azure-keyvault-keys/tests/keys_preparer.py#L27

**Comment by danieljurek** [comment]
`tests.yml` changes LGTM. Someone else should review the other changes.

**Comment by adxsdk6** [comment]
Can one of the admins verify this patch?

**Comment by lmazuel** [comment]
Will do PS preparer from @seankane-msft solving #13987 instead

**Pull request closed without merging**